cmd/operator-sdk/bundle/validate.go

```diff
@@ -143,22 +178,49 @@ func (c bundleValidateCmd) run() (err error) {
 		logger.Info("Unpacked image layers")
 		err = val.PullBundleImage(c.imageTag, dir)
 		if err != nil {
-			logger.Fatalf("Error to unpacking image: %v", err)
+			msg := "Error to unpacking image"
+			if c.json {
+				jsonOutput.addError(fmt.Errorf(msg))
+				jsonOutput.addError(err)
+				fmt.Printf("\nResult:\n%s\n", jsonOutput.toJSON())
+			}
+			logger.Fatalf(msg+": %v", err)
 		}
 	}
 
 	// Validate bundle format.
 	if err = val.ValidateBundleFormat(dir); err != nil {
+		msg := "Bundle format validation failed"
+		if c.json {
+			jsonOutput.addError(fmt.Errorf(msg))
+			for _, err := range err.(bundle.ValidationError).Errors {
```

Comment:
Even though `err` is always `bundle.ValidationError` in the implementation, we cannot know that through type checking statically so use the ok idiom to make sure:
```go
if valErr, isValErr := err.(bundle.ValidationError); isValErr {
	// Use each error as an object
	var outObjs []OutputObject
	for _, err := range valErr.Errors {
		outObjs = append(outObjs, newOutputObject(err.Error())
	}
	fatalJSONObjs(outObjs)
} else {
	// Use single error as an object
	fatalJSON(err.Error())
}
```

Reply:
Currently, it is implemented to only return errors. I mean, all that is returned on it should be considered an error and then, has no warning information. See: https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/bundle/errors.go